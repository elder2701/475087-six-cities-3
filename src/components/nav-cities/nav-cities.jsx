import React, {memo} from "react";
import {ActionCreator} from "../../reducer/city/city.js";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getCities} from "../../reducer/data/selectors.js";
import {getCity} from "../../reducer/city/selectors.js";

const NavCities = ({onChangeCity, cities, activeCity}) => (
  <div className="tabs">
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {cities.map((city) => (
          <li className="locations__item" key={city}>
            <a
              className={`locations__item-link tabs__item ${
                city === activeCity ? `tabs__item--active` : ``
              }`}
              href="#"
              onClick={() => {
                onChangeCity(city);
              }}
            >
              <span>{city}</span>
            </a>
          </li>
        ))}
      </ul>
    </section>
  </div>
);

NavCities.propTypes = {
  onChangeCity: PropTypes.func.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  activeCity: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  cities: getCities(state),
  activeCity: getCity(state)
});

const mapDispatchToProps = (dispatch) => ({
  onChangeCity(city) {
    dispatch(ActionCreator.changeCity(city));
  }
});

export {NavCities};
export default connect(mapStateToProps, mapDispatchToProps)(memo(NavCities));
