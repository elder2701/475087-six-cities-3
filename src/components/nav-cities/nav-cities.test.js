import {NavCities} from "./nav-cities.jsx";
import React from "react";
import renderer from "react-test-renderer";

it(`<NavCities /> sould be render`, () => {
  const tree = renderer
    .create(
        <NavCities
          cities={[`qwe`, `das`]}
          onChangeCity={() => {}}
          getCityOffers={() => {}}
          activeCity={``}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
