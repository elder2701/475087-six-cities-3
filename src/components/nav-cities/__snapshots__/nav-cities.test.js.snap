// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavCities /> sould be render 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            qwe
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            das
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
